vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -232,15 +233,18 @@ synchronized boolean matches(RoutingContext context, String mountPoint, boolean
     if (!enabled) {
       return false;
     }
-    HttpServerRequest request = context.request();
+
+    final HttpServerRequest request = context.request();
+    final MultiMap vertxParams = request.params();
+
     if (!methods.isEmpty() && !methods.contains(request.method())) {
       return false;
     }
     if (path != null && pattern == null && !pathMatches(mountPoint, context)) {
       return false;
     }
     if (pattern != null) {
-      String path = useNormalisedPath ? Utils.normalizePath(context.request().path()) : context.request().path();
```

Comment:
We've just merged a PR switching to the `Utils.normalizePath` call and now back. I'm a bit lost. Can you explain how the previous fix will preserved?